### PR TITLE
Add `.html` to supported file extensions

### PR DESCRIPTION
**Pull request by lifeart**
Resolves https://github.com/ember-template-lint/ember-template-lint/issues/1230

### CODE REVIEW

File: bin/ember-template-lint.js [hunk 1]
```diff
@@ -29,12 +29,13 @@ function lintFile(linter, filePath, toRead, moduleId, shouldFix) {
...
 function expandFileGlobs(positional, ignorePattern) {
   let result = new Set();
+  let supportedExtensions = ['.hbs', '.html'];
 
   positional.forEach((item) => {
     globby
       // `--no-ignore-pattern` results in `ignorePattern === [false]`
       .sync(item, ignorePattern[0] === false ? {} : { ignore: ignorePattern, gitignore: true })
-      .filter((filePath) => filePath.slice(-4) === '.hbs')
+      .filter((filePath) => supportedExtensions.includes(filePath.slice(filePath.lastIndexOf('.'))))
```
Comment (rwjblue) [inline bin/ember-template-lint.js]: Thoughts on using `path.extname` instead of `.lastIndexOf`?

**Comment by rwjblue** [comment]
FWIW, labeled this as breaking per our standard SemVer policy. Since most ember-cli apps are using `ember-template-lint .` they would automatically begin linting additional files that would never have been linted before and therefore _may_ have linting failures.

We’ll need to test this change out against a newly generated ember-cli app to see what is flagged in `app/index.html` and figure out the recommended mitigation policy.

**Comment by lifeart** [comment]
@rwjblue I think we can download fresh app from https://github.com/ember-cli/ember-new-output before each test run, but it need to be discussed about best strategy to implement.

**Comment by rwjblue** [comment]
FYI - I commented over on #1230 with some more actionable steps forward here.

> General path forward:
> 
> 1. Allow consumers to **explicitly** pass globs that include/allow for `.html`. At this stage, calling `ember-template-lint **/*.*` would lint against HTML files but `ember-template-lint .` or `ember-template-lint app/templates` (specifying a non-glob path) would not. This can be released without a major version bump.
> 2. Test to ensure that our linting rules do not flag issues with a newly generated ember-cli app's `index.html`. Fix any issues identified (either with rule changes, or with ember-cli changes).
> 3. Change the default globbing patterns to allow `.html` by default. This is the breaking change, and requires a major version bump.

**Comment by MelSumner** [comment]
@lifeart would you please fix the conflicts? I'm hoping we can remove any blockers and get this in the next major release. Thank you!

File: bin/ember-template-lint.js [hunk 2]
```diff
@@ -45,9 +45,10 @@ function lintSource(linter, options, shouldFix) {
 
 function expandFileGlobs(filePatterns, ignorePattern) {
   let result = new Set();
+  let supportedExtensions = new Set(['.hbs', '.html']);
 
   filePatterns.forEach((pattern) => {
-    let isHBS = pattern.slice(-4) === '.hbs';
+    let isHBS = path.extname(pattern) === '.hbs';
```
Comment (Turbo87) [inline bin/ember-template-lint.js:51]: wouldn't we need to check for `.html` here too?

Comment (lifeart) [reply bin/ember-template-lint.js]: good catch! changes landed

File: bin/ember-template-lint.js [hunk 3]
```diff
@@ -45,12 +45,13 @@ function lintSource(linter, options, shouldFix) {
 
 function expandFileGlobs(filePatterns, ignorePattern) {
   let result = new Set();
+  let supportedExtensions = new Set(['.hbs', '.html']);
```
Comment (rwjblue) [inline bin/ember-template-lint.js:48]: I think we should technically also support `.handlebars` (though it is super uncommonly used, it is still supported by modern ember-cli apps I think).

Comment (lifeart) [reply bin/ember-template-lint.js:48]: It's kinda super legacy, do we have issues to add such support?
(land it - no problem, but remove it - may be in future)

Comment (rwjblue) [reply bin/ember-template-lint.js:48]: Ya, maybe? I don't particularly care to be honest, it costs us very little to support which is why I think we should just mirror whatever ember-cli-htmlbars does by default. 

I also think it might be good to allow this exact list to be configurable in the config file.

Comment (lifeart) [reply bin/ember-template-lint.js:48]: I would suggest to land ability to configure extensions after we land minimal possibility to lint js/ts files, to have stabilyzed logical path to conditionaly apply linting, based on "tricky" extensions

Comment (lifeart) [reply bin/ember-template-lint.js:48]: @rwjblue one note to land `.handlebars` - we have to rewrite more code, because parts relay on 3symbols ext
image

Comment (rwjblue) [reply bin/ember-template-lint.js:48]: What parts rely on 3 chars? We should be using `path.extname`...

Comment (lifeart) [reply bin/ember-template-lint.js]: changes landed https://github.com/ember-template-lint/ember-template-lint/pull/1232/commits/768e9c04fe5ffcf9ff04a1f9060d322f4f44dccc

File: lib/rules/require-iframe-title.js [hunk 4]
```diff
@@ -40,7 +40,7 @@ module.exports = class RequireIframeTitle extends Rule {
           } else if (title.value.type === 'TextNode') {
             let value = title.value.chars.trim();
             let matchIndex = -1;
-            let [, existingNode] = this.knownTitles.filter(([val], index) => {
+            let [, existingNode] = this.knownTitles.find(([val], index) => {
```
Comment (rwjblue) [inline lib/rules/require-iframe-title.js:43]: This change seems unrelated to `.html` files (though it seems fine to do, it should just be a separate PR).

Comment (lifeart) [reply lib/rules/require-iframe-title.js:43]: it was autofixed by `yarn lint:js --fix`

Comment (rwjblue) [reply lib/rules/require-iframe-title.js:43]: O_o why? Seems like that should have caused linting failures on master then, right?

Comment (lifeart) [reply lib/rules/require-iframe-title.js:43]: It may be unicorn rules warning fix

Comment (lifeart) [reply lib/rules/require-iframe-title.js]: reverted autofix
image

**Review by rwjblue — APPROVED** [review]
